lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,137 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], fmt="%g", mode="edge",
```

Comment:
```suggestion
    def bar_label(self, container, captions=[], *, fmt="%g", mode="edge",
```
It's having captions as positional parameters is still clear semantically (`bar_label(container, ['A', 'B', 'C'])`) but all subsequent parameters should be keyword-only.